src/game/logic/object/locomotor.cpp

```diff
@@ -0,0 +1,331 @@
+
+    auto *tmpl = g_theLocomotorStore->Find_Locomotor_Template(key);
+    bool is_new_tmpl = false;
+    if (tmpl == nullptr) {
+        tmpl = NEW_POOL_OBJ(LocomotorTemplate);
+        is_new_tmpl = true;
+        if (ini->Get_Load_Type() == INILoadType::INI_LOAD_CREATE_OVERRIDES) {
+            tmpl->Set_Is_Allocated();
+        }
+    } else {
+        if (ini->Get_Load_Type() == INILoadType::INI_LOAD_CREATE_OVERRIDES) {
+            auto *final_override = static_cast<LocomotorTemplate *>(tmpl->Friend_Get_Final_Override());
+            tmpl = g_theLocomotorStore->New_Override(final_override);
+        }
+    }
+
+    tmpl->Set_Name(tmpl_name);
+
+    ini->Init_From_INI(tmpl, LocomotorTemplate::Get_Field_Parse());
+
+    tmpl->Validate();
+
+    if (is_new_tmpl) {
+        g_theLocomotorStore->m_locomotorTemplates[key] = tmpl;
+    }
+}
+
+// zh: 0x004B7E00 wb: 0x0074E001
+LocomotorTemplate::LocomotorTemplate() :
+    m_surfaces(0),
+    m_maxSpeed(0.0f),
+    m_maxSpeedDamaged(-1.0f),
+    m_minSpeed(0.0f),
+    m_maxTurnRate(0.0f),
+    m_maxTurnRateDamaged(-1.0f),
+    m_acceleration(0.0f),
+    m_accelerationDamaged(-1.0f),
+    m_lift(0.0f),
+    m_liftDamaged(-1.0f),
+    m_braking(99999.0f),
+    m_minTurnSpeed(99999.0f),
+    m_preferredHeight(0.0f),
+    m_preferredHeightDamping(1.0f),
+    m_circlingRadius(0.0f),
+    m_speedLimitZ(999999.0f),
+    m_extra2DFriction(0.0f),
+    m_maxThrustAngle(0.0f),
+    m_behaviorZ(Z_AXIS_BEHAVIOUR_NO_Z_MOTIVE_FORCE),
+    m_appearance(LOCOMOTOR_APPEARANCE_OTHER),
+    m_groupMovementPriority(LOCOMOTOR_GROUP_MOVEMENT_MOVES_MIDDLE),
+    m_accelPitchLimit(0.0f),
+    m_deaccelPitchLimit(0.0f),
+    m_bounceKick(0.0f),
+    m_pitchStiffness(0.1f),
+    m_rollStiffness(0.1f),
+    m_pitchDamping(0.9f),
+    m_rollDamping(0.9f),
+    m_pitchInDirectionOfZVelFactor(0.0f),
+    m_thrustRoll(0.0f),
+    m_thrustWobbleRate(0.0f),
+    m_thrustMinWobble(0.0f),
+    m_thrustMaxWobble(0.0f),
+    m_forwardVelCoef(0.0f),
+    m_lateralVelCoef(0.0f),
+    m_forwardAccelCoef(0.0f),
+    m_lateralAccelCoef(0.0f),
+    m_uniformAxialDamping(1.0f),
+    m_turnPivotOffset(0.0f),
+    m_airborneTargetingHeight(INT_MAX),
+    m_closeEnoughDist(1.0f),
+    m_closeEnoughDist3D(false),
+    m_slideIntoPlaceTime(0.0f),
+    m_locomotorWorksWhenDead(false),
+    m_allowMotiveForceWhileAirborne(false),
+    m_apply2DFrictionWhenAirborne(false),
+    m_downhillOnly(false),
+    m_stickToGround(false),
+    m_canMoveBackwards(false),
+    m_hasSuspension(false),
+    m_maximumWheelExtension(0.0f),
+    m_maximumWheelCompression(0.0f),
+    m_wheelTurnAngle(0.0f),
+    m_wanderWidthFactor(0.0f),
+    m_wanderLengthFactor(1.0f),
+    m_wanderAboutPointRadius(0.0f),
+    m_rudderCorrectionDegree(0.0f),
+    m_rudderCorrectionRate(0.0f),
+    m_elevatorCorrectionDegree(0.0f),
+    m_elevatorCorrectionRate(0.0f)
+{
+}
+
+// zh: 0x004B8070 wb: 0x0074E350
+void LocomotorTemplate::Validate()
+{
+    if (m_maxSpeedDamaged < 0.0) {
```

Comment:
Here value is compared with `double`, and somewhere else is compared with `float`.